### DISCUSSION TITLE
How to close the Samba protocol for users?

### DISCUSSION THREAD

**Discussion by MrHeras555** [SoftEtherVPN/SoftEtherVPN#1621] [Q&A]
How to close Samba protocol for users inside VPN?

Good afternoon! I organize a common local area network between different computers within the country. For my solution I use a separate dedicated server.

Question. When connecting users to our solution, they can use the SAMBA protocol. How can I close the SAMBA protocol for users on the server?

Now all users who use my solution can see network folders (SAMBA) of other computers.

Please tell me how exactly this can be done and can it be done at all?

Thank you.

mail - s.petrosov@lotta-tm.com/pppetrosovv_s@mail.ru

**Comment by metalefty** [comment]
I think you can use ACL. ACL can be applied to group or users.

**Reply by MrHeras555** [reply]
Thank you for answer!) Help pls, which ACL parameters should be applied?

**Reply by metalefty** [reply]
You can specify any destination/src port/address to reject/pass. For example, reject samba server IP address from VPN client. Assemble ACLs as you need.

**Reply by MrHeras555** [reply]
We have a problem. Earlier, with the help of IP Tables, we tried to close port 445 SAMBA, but, unfortunately, VPN SERVER SOFTETER wraps traffic for itself.

**Reply by MrHeras555** [reply]
Could you help pls?

**Reply by chipitsine** [reply]
@MrHeras555 , do you use SecureNAT ?

can you provide your config (with sensitive info curated) ?

**Reply by MrHeras555** [reply]
@chipitsine , Hello! TY for answer.
No, we do not use Secure NAT.
We use ISC-DHCP-SERVER :)
With it, we get an IP address.
Attached config:

declare root
{
	bool CheckServerCert false
	uint64 CreateDateTime 0
	uint64 LastConnectDateTime 0
	bool StartupAccount false
	uint64 UpdateDateTime 0

	declare ClientAuth
	{
		uint AuthType 1
		byte HashedPassword uqlYQcwqrGcLSlGGhkkhmyYLMHw=
		string Username test
	}
	declare ClientOption
	{
		string AccountName game-test-rd
		uint AdditionalConnectionInterval 1
		uint ConnectionDisconnectSpan 0
		string DeviceName VPN
		bool DisableQoS false
		bool HalfConnection false
		bool HideNicInfoWindow false
		bool HideStatusWindow false
		string Hostname 95.85.124.78
		string HubName GAME
		uint MaxConnection 1
		bool NoRoutingTracking false
		bool NoTls1 false
		bool NoUdpAcceleration false
		uint NumRetry 4294967295
		uint Port 443
		uint PortUDP 0
		string ProxyName $
		byte ProxyPassword $
		uint ProxyPort 0
		uint ProxyType 0
		string ProxyUsername $
		bool RequireBridgeRoutingMode false
		bool RequireMonitorMode false
		uint RetryInterval 15
		bool UseCompress false
		bool UseEncrypt true
	}
}

**Reply by chipitsine** [reply]
@MrHeras555 , what is your SE VPN version ? is it 5.XX or 4.XX ?

(they are intended to be interoperable, but sometimes they include their own bugs)

**Reply by MrHeras555** [reply]
Our version SE 4.39.

**Reply by MrHeras555** [reply]
@chipitsine

**Reply by MrHeras555** [reply]
I want to hide SAMBA from client side, between clients. I want samba to not work for them within the range that we provide them.

**Reply by MrHeras555** [reply]
@chipitsine добрый день! Думаю, вам будет удобно общение на русском языке. Как вам кажется, может ли быть проблема связана с тем, что версия сервера может быть забагана? Я это спрашиваю, т.к. мы закрыли порт 445. Цепочка forward была полностью закрыта (политикой drop). Input была тоже закрыта, но, был открыт 443 порт для подключения к серверу. И так же, 22 порт был открыт для подключению к SSH. Возможно, вы знаете, с чем именно может быть проблема?

@chipitsine good afternoon! I think it will be convenient for you to communicate in Russian. Do you think the problem could be related to the server version being bugged? I ask this because we closed port 445. The forward chain was completely closed (by the drop policy). Input was also closed, but port 443 was opened to connect to the server. And also, port 22 was opened for connecting to SSH. Perhaps you know what exactly could be the problem?

**Reply by chipitsine** [reply]
I do not know how "Do you think the problem could be related to the server version being bugged?" is useful for you. The answer is WE DO NOT KNOW (if there are any bugs). But you can also treat it as "yes, I think there might be bugs" if that makes a value to you.

**Reply by chipitsine** [reply]
please provide output of `iptables-save`

**Reply by chipitsine** [reply]
@MrHeras555 , no need to mention me every time. it is getting annoying

**Reply by MrHeras555** [reply]
Attached are the rules that were previously set.
iptables -A INPUT -i ens160 -p tcp --dport 22 -m state --state NEW,ESTABLISHED -j ACCEPT
iptables -A OUTPUT -o ens160 -p tcp --sport 22 -m state --state ESTABLISHED -j ACCEPT
iptables -P INPUT DROP
iptables -P FORWARD DROP
iptables -A FORWARD -p tcp --dport 445 -i tap_vpn -j DROP

But it didn't work out for them.

**Reply by MrHeras555** [reply]
> @MrHeras555 , no need to mention me every time. it is getting annoying

Sorry, 1st time in github.

**Reply by chipitsine** [reply]
are those iptables rules added by you manually ? if so, your question is more about managing iptables

**Reply by MrHeras555** [reply]
> Are those iptables rules added by you manually ? if so, your question is more about managing iptables

Yes we added that rules. We manually added these IPTables rules. we want to manually close the transfer of files using the SAMBA protocols within the local network.

i.e., when a user connects to the server, they get a gray IP. And when another user connects to the server, he also gets a gray IP.

Our goal is to close the possibility of transferring files via SAMBA protocols between users within the local network :)

**Reply by chipitsine** [reply]
if you see your rules in `iptables-save`, you are almost there.

I would suggest the following as iptables troubleshoot `-j LOG --log-prefix='[sevpn] '`, i.e. the same rule as "DROP", but with `LOG` instead.  You can see whether packets hit your rule or do not hit.

I suggest to try `OUTPUT` instead of `FORWARD` chain, but you play with various rules keeping in mind `LOG` approach

**Reply by chipitsine** [reply]
modern SMB still use 445, but I would have a look at wireshark, maybe they switched to some other port
